File: Java/src/_0011_Container_With_Most_Water/Solution.java
package _0011_Container_With_Most_Water;

/**
 * 暴力解法
 * 时间复杂度：O(n^2)
 * 空间复杂度：O(1)
 */
public class Solution {

    public int maxArea(int[] height) {

        int res = Math.min(height[0], height[1]);
        for (int i = 0; i < height.length; i++)
            for (int j = i + 1; j < height.length; j++)
                res = Math.max(res, Math.min(height[i], height[j]) * (j - i));

        return res;
    }

    public static void main(String[] args) {

        int[] height = {1, 8, 6, 2, 5, 4, 8, 3, 7};
        System.out.println((new Solution()).maxArea(height));
    }
}

File: Python/_0011_Container_With_Most_Water/Solution.py
# coding=utf-8
from typing import List


class Solution:
    def maxArea(self, h: List[int]) -> int:
        res = 0
        i = 0; j = len(h) - 1
        while i < j:
            res = max(res, min(h[i], h[j]) * (j - i))
            if h[i] > h[j]:
                j -= 1
            else:
                i += 1
        return res


if __name__ == "__main__":
    height = [1, 8, 6, 2, 5, 4, 8, 3, 7]
    print(Solution().maxArea(height))  # 49
